### ISSUE TITLE
Create a dataset loader for MASH-QA

### ISSUE THREAD

**Issue by hakunanatasha** [bigscience-workshop/biomedical#56]
From https://github.com/mingzhu0527/MASHQA

**Comment by ruisi-su** [comment]
I added the loading script for this which I created by following the previous iteration of the hackathon doc.

**Comment by vasudev-sharma** [comment]
Interested.

**Comment by tanmaylaud** [comment]
#self-assign

**Comment by hakunanatasha** [comment]
Hi @tanmaylaud 

Have a look at the following thread and let me know if it's useful: https://github.com/huggingface/datasets/issues/418
google drive links should be able to work. I think you have another dataset pending with the same blocker too; hopefully this helps.

@ruisi-su has also worked with Mash-QA before, so ping us for any assistance.

**Comment by tanmaylaud** [comment]
@ruisi-su where have you added the loading script you mentioned above? thanks for the help!

**Comment by tanmaylaud** [comment]
> @ruisi-su where have you added the loading script you mentioned above? thanks for the help!

@ruisi-su Could you please let me know about this?

**Comment by tanmaylaud** [comment]
> @tanmaylaud how about the below:
> 
> ```python
>     # downloading from google drive solution: https://github.com/huggingface/datasets/issues/418
>     def _get_drive_url(self, url):
>         base_url = "https://drive.google.com/uc?id="
>         split_url = url.split("/")
>         return base_url + split_url[5]
> ```
> 
> Then in your _split_generators():
> 
> ```python
>     def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
>         """
>         Step 5: Download and extract the dataset.
> 
>         For each config name, run `download_and_extract` from the dl_manager;
>         this will download and unzip any files within a cache directory,
>         specified by `data_dir`.
> 
>         `download_and_extract` can accept an iterable object and return the same structure with the url replaced with the path to local files:
> 
>         ex: output = dl_manager.download_and_extract(
>         {"data1:" "url1", "data2": "url2"})
> 
>         output
>         >> {"data1": "path1", "data2": "path2"}
> 
>         Nested zip files can be cached also, but make sure to save their
>         path.
> 
>         Populate "SplitGenerator" with `name` and `gen_kwargs`. Note:
> 
>         - `name` can be: datasets.Split.<TRAIN/TEST/VALIDATION> or a string
>         - all keys in `gen_kwargs` can be passed to `_generate_examples()`. If your dataset has multiple files, you can make a separate key for each file, as shown below:
> 
>         """
> 
>         urls = _URLS[_DATASETNAME]
>         data_dir = os.path.join(dl_manager.download_and_extract(self._get_drive_url(urls)), _DATA_DIR)
>  <continuation of the rest of the function body>
> ```
> 
> Note that there was a `join` on the path because the google drive link actually stores the dataset in a subfolder called `mashqa_data`. Feel free to drop that if you find a better way to get the `data_dir`.

Hey, 
thanks but my doubt was about the data schema. The format is slightly different from standard SQUAD format which is a bit confusing. I thought you had worked on the dataset?

**Comment by ruisi-su** [comment]
@tanmaylaud I worked on this dataset in previous alpha runs (before the BigBio schema was finalized), which is why the loading script is not publicly available. When you said the format is slightly different from standard SQUAD, could you be more specific? Are you referring to the [BigBio schema for qa](https://github.com/bigscience-workshop/biomedical/blob/master/utils/schemas/qa.py)?

**Comment by tanmaylaud** [comment]
> @tanmaylaud I worked on this dataset in previous alpha runs (before the BigBio schema was finalized), which is why the loading script is not publicly available. When you said the format is slightly different from standard SQUAD, could you be more specific? Are you referring to the [BigBio schema for qa](https://github.com/bigscience-workshop/biomedical/blob/master/utils/schemas/qa.py)?

No, I am referring to the MASHQA dataset itself. The overall structure is confusing> I reached out to the author but didn't get a response.

**Comment by ruisi-su** [comment]
@tanmaylaud I am a bit confused on this. The MASHQA dataset seems to follow the SQUAD format (consistent dictionary structures of `paragraphs`, `qas`, `answers`, `questions`, and `context`. I followed the [data loader](https://github.com/huggingface/datasets/blob/master/datasets/squad/squad.py) for the SQUAD dataset when I worked on this earlier and did not run into any issues. Could you be more specific about the confusion here?
